packages/remix-dev/__tests__/routesConvention-test.ts

```diff
@@ -36,6 +36,47 @@ describe("createRoutePath", () => {
       ["[index]", "index"],
       ["test/inde[x]", "test/index"],
       ["[i]ndex/[[].[[]]", "index/[/[]"],
+
+      // Optional segment routes
+      ["(routes)/$", "routes?/*"],
+      ["(routes)/($)", "routes?/*?"], // TODO: Fails, do we want to allow this?
+      ["(routes)/(sub)/$", "routes?/sub?/*"],
+      ["(routes).(sub)/$", "routes?/sub?/*"],
+      ["(routes.sub)/$", "routes?/sub?/*"], // TODO: Fails, do we want to allow this?
```

Comment:
Personally, I don't think we need to add the complexity of being able to interpolate dots inside an optional segment - the line above is how I'd expect optional segments + dot-notation to work together.  In this case I'd expect the generated React Router path to be `<Route path="routes.sub?/*">`

Reply:
```
["(routes.sub)/$", "routes?/sub?/*"] // do we want to allow this?
```

The word "routes" there is confusing to me, so I'm changing it to `"products.all"` for my comment 😅

```
["(products.all)/$", "products?/all?/*"] // do we want to allow this?
```

No, we don't. The React Router version has no way of "grouping" two segments as optional together:

```jsx
// this is only for "sub"
<Route path="foo/bar?" />

// this is for both
<Route path="routes?/sub?" />
```

So in Remix it's the same, only individual segments can be optional:

```
(products).(all).tsx -> products?/all?
```